### DISCUSSION TITLE
Poweroff from shell?

### DISCUSSION THREAD

**Discussion by Lexxior** [Hypfer/Valetudo#2029] [Q&A / Support] [answered]
### Are you already using Valetudo?

Yes, for a few months

### Your questions

Hello!
I use my S5 Max only 3 times per week and would like to turn it off between uses to preserve battery life. This works fine with the button but via the shell it does not work. If I run _shutdown_ or _/sbin/shutdown_ the robo disconnects from the wifi, the lights stay on and after a couple of minutes it turns itself on again.
It's not connected to the charger
Does anyone know how to achieve this or mimic a press on the power button?

**Comment by Hypfer** [comment, marked as answer]
> If I run shutdown or /sbin/shutdown the robo disconnects from the wifi, the lights stay on and after a couple of minutes it turns itself on again.

Correct. `shutdown` just shuts down the linux side of the robot. The MCU side stays awake.
Your S5 Max actually handles that quite gracefully and automatically boots back up again. Other robots may become softbricked until a hard power cycle if you do that. Please don't run `shutdown`.

Recently, however, I was made aware that there is _some_ possibility, as on various roborocks, the `uart_test` binary allows telling the MCU to shut down the robot. There _might_ be unexpected side-effects from doing that though. It is definitely not used in normal operation and with that, it might be unexpectedly broken.
You can also **permanently and unrecoverably brick your robot** by playing around with various `uart_test` commands so **do not do that**.

That said, I just ran `killall rr_loader && sync && sleep 2 && uart_test -m s` on my S5 Max and it seems to have done the trick.

It is however very much possible that this will be harmful further down the line. Maybe some caches don't get flushed and with that data corruption happens. Or maybe there's some watchdog monitoring the system that gets triggered by `rr_loader` stopping, might set a `failed` flag or something and then reset the robot or maybe softbrick it.

Because of all these unknowns and since the power and battery savings are minimal anyway, I'd strongly advise against doing this.
You can do it, but you won't receive any further support for it or anything that might have been caused by that.
